src/EFCore.PG/Migrations/NpgsqlMigrationsSqlGenerator.cs

```diff
@@ -672,12 +672,16 @@ protected override void Generate(RenameTableOperation operation, IModel model, M
 
         protected override void IndexOptions(CreateIndexOperation operation, IModel model, MigrationCommandListBuilder builder)
         {
-            if (operation[NpgsqlAnnotationNames.IndexInclude] is string[] includeColumns && includeColumns.Length > 0)
+            if (_postgresVersion.AtLeast(11))
             {
-                builder
-                    .Append(" INCLUDE (")
-                    .Append(ColumnList(includeColumns))
-                    .Append(")");
+                if (operation[NpgsqlAnnotationNames.IndexInclude] is string[] includeColumns &&
```

Comment:
The nested condition can be safely merged with the outer since there are no returns on `else` blocks.

Reply:
Done.

Note that I wasn't totally sure here about something - would be good to have your opinion (also @bricelam's).

In most cases, if something isn't supported, I prefer throwing an exception to make it clear to the user. In the case of index-included properties, for old versions which don't support this, for now I've decided to simply ignore - creating the index without the included properties. This is because included properties are an optional feature which only improves perf, and this makes the same model usable on any version of PG: with newer versions we get the speedup, with older versions we don't.

Does this make sense? Or is it better to throw on older versions?

Reply:
Throwing or ignoring both seem fine to me. Did you consider logging a warning?

Reply:
I keep forgetting those :) Ignoring and logging does sound like the way to go.